test/integration/query-interface.test.js

```diff
@@ -649,6 +649,56 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         });
         let constraints = await this.queryInterface.showConstraint('posts');
         constraints = constraints.map(constraint => constraint.constraintName);
+        expect(constraints).to.include('posts_username_users_fk');
+
+        if (dialect === 'postgres') {
+          const foreignKeys = await this.sequelize.query(
+            this.queryInterface.queryGenerator.getForeignKeysQuery('posts', this.sequelize.config.database),
+            { type: this.sequelize.QueryTypes.FOREIGNKEYS },
+          );
+          expect(Object.keys(foreignKeys[0])).to.have.length(8);
+          expect(foreignKeys[0].condef).to.equal('FOREIGN KEY (username) REFERENCES users(username) ON UPDATE CASCADE ON DELETE CASCADE');
+        }
```

Comment:
I don't think this change is necessary. The unit test already checks that the SQL is generated without `NOT VALID`